cbopensource/connectors/taxii/cybox_parse.py

```diff
@@ -201,5 +254,27 @@ def cybox_parse_observable(observable, indicator, timestamp, score):
                                 'timestamp': timestamp,
                                 'link': link,
                                 'score': score})
+        if props.sha256:
+            iocs = {'sha256': []}
+            if type(props.sha256) is list:
+                for sha256 in props.sha256:
+                    if validate_sha256(sha256):
+                        iocs['sha256'].append(sha256.strip())
+            else:
+                if hasattr(props.sha256, 'value'):
+                    sha256 = props.sha256.value.strip()
+                else:
+                    sha256 = props.sha256.strip()
+                if validate_sha256(sha256):
+                    iocs['sha256'].append(sha256)
+
+            if len(iocs['sha256']) > 0:
+                reports.append({'iocs': iocs,
```

Comment:
This is being done over and over again.  That tells me that there is probably an easier way to do this.